pkg/metrics/metrics.go

```diff
@@ -256,25 +215,62 @@ func isEndpointsValid(nodes *v1.NodeList, endpoints *v1.Endpoints) bool {
 	return true
 }
 
-// updateServiceMonitors patches the metrics Service Monitor to include required fields to display node graphs on the
-// OpenShift console. Console graph queries require metrics endpoint target name to be node name, however
-// windows_exporter returns node IP. We replace the target name by adding `replace` action field to the ServiceMonitor
-// object that replaces node IP to node name as the metrics endpoint target.
-func updateServiceMonitors(cfg *rest.Config, namespace string) error {
-
-	patchData := fmt.Sprintf("[{\"op\": \"replace\", \"path\": \"/spec/endpoints/0\", "+
-		"\"value\":{\"path\": \"/%s\",\"port\": \"%s\",\"relabelings\": [{\"action\": \"replace\", \"regex\": \"(.*)\", "+
-		"\"replacement\": \"$1\", \"sourceLabels\": [\"__meta_kubernetes_endpoint_address_target_name\"],"+
-		"\"targetLabel\": \"instance\"}]}}]", PortName, PortName)
+// Configure takes care of all the required configuration steps
+// for Prometheus monitoring like validating monitoring label
+// and creating metrics Endpoints object.
+func (c *Config) Configure(ctx context.Context) error {
+	// validate if cluster monitoring is enabled in the operator namespace
+	enabled, err := c.validate(ctx)
+	if err != nil {
+		return errors.Wrap(err, "error validating cluster monitoring label")
+	}
+	// Create Metrics Endpoint object only if monitoring is enabled
+	if enabled {
+		if err := c.createEndpoint(); err != nil {
+			return errors.Wrap(err, "error creating metrics Endpoint")
+		}
+	}
+	return nil
+}
```

Comment:
nit: You can reduce indentation by handling the not enabled case first
Either approach is fine since the enabled case is fairly short in this case
```

	enabled, err := c.validate(ctx)
	if err != nil {
		return errors.Wrap(err, "error validating cluster monitoring label")
	}
	// Create Metrics Endpoint object only if monitoring is enabled
	if !enabled {
		return nil
	}
	if err := c.createEndpoint(); err != nil {
		return errors.Wrap(err, "error creating metrics Endpoint")
	}
	return nil
}

```